hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieClusteringConfig.java

```diff
@@ -732,40 +712,17 @@ public String getValue() {
     }
   }
 
+  @EnumDescription("Clustering mode to use.")
   public enum ClusteringOperator {
 
-    /**
-     * only schedule the clustering plan
-     */
-    SCHEDULE("schedule"),
-
-    /**
-     * only execute then pending clustering plans
-     */
-    EXECUTE("execute"),
-
-    /**
-     * schedule cluster first, and execute all pending clustering plans
-     */
-    SCHEDULE_AND_EXECUTE("scheduleandexecute");
+    @EnumFieldDescription("Only schedule the clustering plan.")
+    SCHEDULE,
 
-    private static final Map<String, ClusteringOperator> VALUE_TO_ENUM_MAP =
-            TypeUtils.getValueToEnumMap(ClusteringOperator.class, e -> e.value);
+    @EnumFieldDescription("Only execute pending clustering plans.")
+    EXECUTE,
 
-    private final String value;
-
-    ClusteringOperator(String value) {
-      this.value = value;
-    }
-
-    @Nonnull
-    public static ClusteringOperator fromValue(String value) {
-      ClusteringOperator enumValue = VALUE_TO_ENUM_MAP.get(value);
-      if (enumValue == null) {
-        throw new HoodieException(String.format("Invalid value (%s)", value));
-      }
-      return enumValue;
-    }
+    @EnumFieldDescription("Schedule cluster first, and execute all pending clustering plans.")
+    SCHEDULE_AND_EXECUTE;
```

Comment:
Do we still need to keep the naming here: `SCHEDULE_AND_EXECUTE("scheduleandexecute")`? Because `"scheduleandexecute"` cannot be converted from `SCHEDULE_AND_EXECUTE` by just switching the case.

Reply:
We need to make sure the refactoring is still backward compatible.

Reply:
Yeah, I reverted it. I mentioned it in the jira issue I created